File: lua/hello.lua
HelloWorld = {
    run = function()
        io.write("Hello, ", HelloWorld.get_name(), "\n")
    end,

    get_name = function()
        repeat
            io.write("Please enter your name: ")
            name = io.read()
        until name ~= ""
        return name
    end
}


h = HelloWorld
h.run()

File: go/hello.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

func GetName() string {
    scanner := bufio.NewScanner(os.Stdin)
    for {
        fmt.Print("Enter your name: ");
        scanner.Scan()
        name := strings.TrimSpace(scanner.Text())
        if (name != "") {
            return name
        }
    }
}

func main() {
    fmt.Println("Hello, " + GetName())
}
